### PR TITLE
001 job roles front end

### PR DESCRIPTION
**Pull request by Adnanhoq**

### CODE REVIEW

File: views/jobRoles.njk [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
```
Comment (Leebee1392) [inline views/jobRoles.njk:4]: Remove unnecessary screenshot from ticket

File: src/app.ts [hunk 2]
```diff
@@ -37,4 +41,5 @@ app.get('/', async (req: express.Request, res: express.Response): Promise<void>
   res.render('index.njk');
 });
 
-app.get('/test', getAllDatabases);
\ No newline at end of file
+app.get('/test', getAllDatabases);
```
[the remark this answers is not in the document]
Comment (Adnanhoq) [reply src/app.ts:44]: Im trying to merge first 😂

File: src/models/JobRoleResponse.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export type JobRoleResponse = {
+    roleId: number,
+    roleName: String,
+    locations: String,
```
[the remark this answers is not in the document]
Comment (Adnanhoq) [reply src/models/JobRoleResponse.ts:4]: No, as the SQL query returns it as a string

File: src/models/JobRoleResponse.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+    capability: String,
```
[the remark this answers is not in the document]
Comment (Adnanhoq) [reply src/models/JobRoleResponse.ts:5]: No, as the SQL query returns it as a string

File: src/models/JobRoleResponse.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+    band: String,
```
[the remark this answers is not in the document]
Comment (Adnanhoq) [reply src/models/JobRoleResponse.ts:6]: No, as the SQL query returns it as a string

File: src/services/JobRoleService.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import axios, { AxiosResponse } from "axios";
+import { JobRoleResponse } from "../models/JobRoleResponse";
+
+axios.defaults.baseURL = process.env.API_URL || 'http://localhost:8080/';
```
[the remark this answers is not in the document]
Comment (Adnanhoq) [reply src/services/JobRoleService.ts:4]: Done

File: views/jobRoles.njk [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+<body>
+    <div class="container">
+        {% include "components/navbar.njk" %}
+
+        <div class="mainBody">
+            <h1>Job Roles List</h1>
+            <div class="row">
+                {% for role in jobRole | sort(attribute='roleId') %}
+                    <div class="col-md-4 mb-4">
+                    <a href = "./jobRoleDetail" target="_blank" class="jobCard">
```
[the remark this answers is not in the document]
Comment (Adnanhoq) [reply views/jobRoles.njk:21]: Fixed

File: views/jobRoles.njk [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+                        <div class="card">
+                            <div class="card-body">
```
[the remark this answers is not in the document]
Comment (Adnanhoq) [reply views/jobRoles.njk:23]: css doesn't like me, i'll try again

File: views/jobRoles.njk [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+                                <h5 class="card-title">{{ role.roleName }}</h5>
+                                <p class="card-text"><strong>Locations:</strong> {{ role.locations }}</p>
+                                <p class="card-text"><strong>Capability:</strong> {{ role.capability }}</p>
+                                <p class="card-text"><strong>Band:</strong> {{ role.band }}</p>
+                                <p class="card-text"><strong>Closing Date:</strong> {{ role.closingDate | date }}</p>
```
[the remark this answers is not in the document]
Comment (Adnanhoq) [reply views/jobRoles.njk:28]: I have done this in filters folder

File: views/jobRoles.njk (hunk 1 above)
Comment (Adnanhoq) [reply views/jobRoles.njk:4]: Done

File: src/app.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Adnanhoq) [reply src/app.ts:44]: I can't change this as I don't have his files or dependencies

File: src/models/JobRoleResponse.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Leebee1392) [reply src/models/JobRoleResponse.ts:4]: Your database design has enums so it should match your code no matter what the database returns

File: src/models/JobRoleResponse.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Leebee1392) [reply src/models/JobRoleResponse.ts:5]: Your database design has enums so it should match your code no matter what the database returns

File: src/models/JobRoleResponse.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Leebee1392) [reply src/models/JobRoleResponse.ts:6]: Your database design has enums so it should match your code no matter what the database returns

File: test/unit/controllers/JobRoleControllerTests.ts
Comment (Leebee1392) [inline test/unit/controllers/JobRoleControllerTests.ts:1]: This is still empty?

File: views/jobRoles.njk (hunk 8 above)
[the remark this answers is not in the document]
Comment (Leebee1392) [reply views/jobRoles.njk:21]: I can't see a new screenshot so I can't verify

File: views/jobRoles.njk (hunk 9 above)
[the remark this answers is not in the document]
Comment (Leebee1392) [reply views/jobRoles.njk:26]: I can't see a new screenshot so I can't verify

File: src/services/JobRoleService.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import config from "../config";
+import axios, { AxiosResponse } from "axios";
+import { JobRoleResponse } from "../models/JobRoleResponse";
+
+export const URL: string = "/api/job-roles/";
```
Comment (Leebee1392) [inline src/services/JobRoleService.ts:5]: I don't think this is correct

**Review by Leebee1392 — CHANGES_REQUESTED** [review]
Are you sure this meets requirements of the ticket?

File: src/controllers/JobRoleController.ts [hunk 11]
```diff
@@ -0,0 +1,8 @@
+import express from "express";
+import { getJobRoles } from "../services/JobRoleService";
+
+export const getAllJobRoles = async (req: express.Request, res: express.Response): Promise<void> => {
+    const jobRole = await getJobRoles();
+    console.log(jobRole)
```
Comment (Leebee1392) [inline src/controllers/JobRoleController.ts:6]: Needs to be taken out

File: src/models/JobRoleResponse.ts [hunk 12]
```diff
@@ -0,0 +1,8 @@
+export interface JobRoleResponse {
+    roleId: number;
+    roleName: string;
+    locations: string;
```
Comment (Leebee1392) [inline src/models/JobRoleResponse.ts:4]: Make these enums

File: src/models/JobRoleResponse.ts [hunk 13]
```diff
@@ -0,0 +1,8 @@
...
+    capability: string;
```
Comment (Leebee1392) [inline src/models/JobRoleResponse.ts:5]: Make these enums

File: src/models/JobRoleResponse.ts [hunk 14]
```diff
@@ -0,0 +1,8 @@
...
+    band: string;
```
Comment (Leebee1392) [inline src/models/JobRoleResponse.ts:6]: Make these enums

File: src/models/JobRoleResponse.ts (hunk 12 above)
Comment (Leebee1392) [inline src/models/JobRoleResponse.ts:4]: Should this be an array?

File: src/services/JobRoleService.ts [hunk 15]
```diff
@@ -0,0 +1,14 @@
+import axios, { AxiosResponse } from "axios";
+import { JobRoleResponse } from "../models/JobRoleResponse";
+
+export const JOBROLEURL = "/job-roles/";
+
+export const getJobRoles = async (): Promise<JobRoleResponse[]> => {
+    try{
+        const response: AxiosResponse<JobRoleResponse[]> = await axios.get("http://localhost:8080/api/job-roles");
```
Comment (Leebee1392) [inline src/services/JobRoleService.ts:8]: No hardcoded URLs

File: tsconfig.json [hunk 16]
```diff
@@ -12,7 +12,8 @@
     },
     "include": [
         "src/**/*",
-        "./.env"
+        "test/**/*",
+        "./.env",
```
Comment (Leebee1392) [inline tsconfig.json:16]: Should not have either of these in the include section

File: views/jobRoles.njk (hunk 9 above)
[the remark this answers is not in the document]
Comment (Leebee1392) [reply views/jobRoles.njk:26]: The date still isn't correct

File: src/services/TestService.ts [hunk 17]
```diff
@@ -6,6 +6,7 @@ export const getDatabases = async (): Promise<string[]> => {
         console.log(config.API_URL);
```
Comment (Leebee1392) [inline src/services/TestService.ts:6]: Remove log

File: src/services/TestService.ts [hunk 18]
```diff
@@ -6,6 +6,7 @@ export const getDatabases = async (): Promise<string[]> => {
...
         const response: AxiosResponse<string[]> = await axios.get(config.API_URL+"test");
 
+
```
Comment (Leebee1392) [inline src/services/TestService.ts:9]: Remove unnecessary line

**Review by Leebee1392 — CHANGES_REQUESTED** [review]
Your configuration is breaking the accessibility tests and will probably break other tests in the future.

image

**Review by Leebee1392 — APPROVED** [review]

**Review by Leebee1392 — APPROVED** [review]

**Review by Leebee1392 — CHANGES_REQUESTED** [review]
Merge conflicts

**Review by Leebee1392 — APPROVED** [review]

**Review by AZZohaib — APPROVED** [review]

**Pull request merged**